packages/branch-protector/src/index.ts

```diff
@@ -48,12 +50,26 @@ async function protectBranch(
 	}
 }
 
+function createEvents(messages: Message[]): UpdateBranchProtectionEvent[] {
+	const res = messages
+		.map((msg) => msg.Body)
+		.filter((msg): msg is string => !!msg)
+		.map((msg) => JSON.parse(msg) as UpdateBranchProtectionEvent);
+
+	return res;
+}
+
 export async function main() {
 	const config: Config = await getConfig();
 	const octokit: Octokit = await getGithubClient(config);
+	const sqsClient = new SQSClient({});
 
-	const events = await getEvents(1, config);
-	for (const event of events) {
-		await protectBranch(octokit, config, event);
-	}
+	const messages = await readFromQueue(config, 1, sqsClient);
+	const events = createEvents(messages);
+	await Promise.all(
+		events.map(async (event) => await protectBranch(octokit, config, event)),
```

Comment:
```suggestion
		events.map((event) => protectBranch(octokit, config, event)),
```

If any call to protectBranch fails (Github API outage seems pretty common these days!) would this mean that events that were handled successfully won't be removed from the queue?

Reply:
yes. good point!